### PR TITLE
Improve segment metadata fetch API

### PR DESCRIPTION
**Pull request by yupeng9**
To support remotely pushed segment in upsert tables.

### CODE REVIEW

**Review by Jackie-Jiang — COMMENTED** [review]
We can simplify the logic to:
- If the input table name has suffix, directly use it
- If not, try both OFFLINE and REALTIME, and return the result if metadata for the input segment exists

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java [hunk 1]
```diff
@@ -290,9 +290,23 @@
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
       @ApiParam(value = "Name of the segment", required = true) @PathParam("segmentName") @Encoded String segmentName) {
     segmentName = URIUtils.decode(segmentName);
-    TableType tableType = SegmentName.isRealtimeSegmentName(segmentName) ? TableType.REALTIME : TableType.OFFLINE;
-    String tableNameWithType =
-        ResourceUtils.getExistingTableNamesWithType(_pinotHelixResourceManager, tableName, tableType, LOGGER).get(0);
+    String tableNameWithType = tableName;
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java:293]: What I meant is to directly try reading the metadata without checking the table existence
```
    Map<String, String> segmentMetadata = null;
    if (TableNameBuilder.getTableTypeFromTableName(tableName) != null) {
      segmentMetadata = getSegmentMetadataInternal(tableName, segmentName);
    } else {
      segmentMetadata = getSegmentMetadataInternal(TableNameBuilder.OFFLINE.tableNameWithType(tableName), segmentName);
      if (segmentMetadata == null) {
        segmentMetadata = getSegmentMetadataInternal(TableNameBuilder.REALTIME.tableNameWithType(tableName), segmentName);
      }
    }
    if (segmentMetadata != null) {
      ...
```

**Review by Jackie-Jiang — COMMENTED** [review]
Can we add a test for this?

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java (hunk 1 above)
Comment (yupeng9) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java:293]: ah, I see what you meant now

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**